game/camera.cpp

```diff
@@ -122,6 +125,32 @@ void Camera::setMode(Camera::Mode m) {
     dst.spin.y = pl->rotation();
   }
 
+void Camera::setMarvinMode(Camera::MarvinMode m) {
+  if(camMarvinMod==m)
+    return;
+  auto pl = Gothic::inst().player();
+  if(pl==nullptr)
+    return;
+  if(m==M_Free)
+    pl->setAnim(AnimationSolver::NoAnim);
+  if(m==M_Fixed) {
+    auto offset    = dst.target - pl->cameraBone();
+    float k        = float(M_PI/180.0);
+    float s        = std::sin(dst.spin.y*k), c=std::cos(dst.spin.y*k);
+    cameraOffset.x = -offset.x * c - offset.z * s;
+    cameraOffset.z = offset.x * s - offset.z * c;
+    cameraOffset.y = offset.y;
+    }
+  else if(camMarvinMod==M_Fixed) {
```

Comment:
This line is misleading.
2 previous condition in a row were about `m`(new mode), suggesting switch-case like workflow, but here it's suddenly `camMarvinMod`.